sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -195,6 +213,385 @@ describe("StudentAppealStudentsController(e2e)-submitStudentAppeal", () => {
+          pdDependentUpload: [
+            {
+              storage: "url",
+              originalName: "JessLee_AlexLeeTax_2023.txt",
+              name: "JessLee_AlexLeeTax_2023-4315285a-b756-4084-95dc-781b51692d27.txt",
+              url: "student/files/JessLee_AlexLeeTax_2023-4315285a-b756-4084-95dc-781b51692d27.txt",
+              size: 0,
+              type: "text/plain",
+              hash: "1cb251ec0d568de6a929b520c4aed8d1",
+            },
+          ],
+        },
+      ],
+      supportnocustodyDependants: "yes",
+      dependantCustodyFileUpload: [
+        {
+          storage: "url",
+          originalName: "JessLee_DependantsCustody.txt",
+          name: "JessLee_DependantsCustody-b76efea7-1515-4bd4-9a90-e55d7dde05a0.txt",
+          url: "student/files/JessLee_DependantsCustody-b76efea7-1515-4bd4-9a90-e55d7dde05a0.txt",
+          size: 0,
+          type: "text/plain",
+          hash: "1cb251ec0d568de6a929b520c4aed8d1",
+        },
+      ],
+    };
+    const payload: StudentAppealAPIInDTO = {
+      studentAppealRequests: [
+        {
+          formName: DEPENDANT_INFORMATION_FORM_NAME,
+          formData: dependantInformationData,
+        },
+      ],
+    };
+    // Mock user service to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+    // Get any student user token.
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Mock the form service to validate the dry-run submission result.
+    // TODO: Form service must be hosted for E2E tests to validate dry run submission
+    // and this mock must be removed.
+    const formService = await getProviderInstanceForModule(
+      appModule,
+      AppStudentsModule,
+      FormService,
+    );
+    const dryRunSubmissionMock = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: FINANCIAL_INFORMATION_FORM_NAME,
+      data: { data: dependantInformationData },
+    });
+
+    formService.dryRunSubmission = dryRunSubmissionMock;
+
+    const endpoint = `/students/appeal/application/${application.id}`;
+
+    // Act/Assert
+    const response = await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.CREATED);
+
+    const newStudentAppealRequest = await studentAppealRequestRepo.findOne({
+      where: { studentAppeal: { id: response.body.id } },
+    });
+    expect(newStudentAppealRequest.submittedData).toStrictEqual(
+      payload.studentAppealRequests[0].formData,
+    );
+
+    // Expect to call the dry run submission.
+    expect(dryRunSubmissionMock).toHaveBeenCalledWith(
+      DEPENDANT_INFORMATION_FORM_NAME,
+      {
+        ...dependantInformationData,
+        programYear: application.programYear.programYear,
+      },
+    );
+  });
+
+  it(
+    "Should determine the total number of student appeal requests for a student appeal when the student " +
```

Comment:
I would say that we have to keep student income and partner income appeal out of the tests totally as they are not built on real requirements.